samples/quickperf/readme.md

```diff
@@ -0,0 +1,308 @@
+* Sampling of records for seeding foreign-key relationships or testing against a specific subset of data
+* Batch mode support
+* Automatic statement and transaction tagging
+
+
+# Installation on Ubuntu 
+```
+sudo apt-get install openjdk-8-jdk
+sudo apt install maven
+```
+
+## Authentification 
+It is recommended to use a service account, otherwise larger scale tests will run into quota limitations
+
+Set active auth to service account:
+```
+gcloud auth list
+gcloud config set account xxx-compute@developer.gserviceaccount.com
+```
+
+# Configuration
+
+## Parameters
+```
+{
+    "project": "Project ID",
+    "instance": "Spanner Instance",
+    "database": "Spanner Database",
+    "threads": Number of concurrent threads,
+    "iterations": Number of how often a statement should be executed in a thread,
+    "query": "Statement (e.g. query)",
+    "samplingQuery": "OPTIONAL - Sampling query",
+    "writeMetricToFile": Will write latency metrics to a file (true/false),
+    "batchSize": If testing batching - determines how large a batch size would be,
+    "queryParams": [
+        {"order": 1, "value": "query paramters with value generator"}
+    ]
+}
+```
+
+## Example Config
+```
+{
+    "project": "xxxx",
+    "instance": "xxxx",
+    "database": "users",
+    "threads": 1,
+    "iterations": 10,
+    "query": "SELECT users.user_id, membership.enrolled, GroupMgmt.grpname FROM users, GroupMgmt, membership WHERE users.user_id = ? AND users.user_id = membership.user_id AND GroupMgmt.group_id = membership.group_id",
+    "samplingQuery": "SELECT user_id FROM Users TABLESAMPLE RESERVOIR (100000 ROWS)",
+    "writeMetricToFile": false,
+    "queryParams": [
+        {"order": 1, "value": "#pi"}
+    ]
+}
+```
+
+# Hello World Example
+
+The folder `exampleconfigs/config.json` contains a simple setup that runs SELECT 1 against the database
+
+Configure the right Spanner `project` and `instance` and run the app. 
+
+**config.json**
+```
+{
+"project": "xxxx",
+"instance": "xxx",
+"database": "xxx",
+"threads": 1,
+"iterations": 100,
+"query": "SELECT 1",
+"writeMetricToFile": false,
+"batchSize": 0
+}
+```
+
+**Run:**
+```
+mvn -q exec:java -Dexec.mainClass="com.google.cloud.jdbc.quickperf.QuickPerf" \
+-Dexec.args="-c exampleconfigs/config.json"
+```
+
+
+
+# End-to-End Example
+
+Generates three tables with n:m relationships and performs a load test.
+
+All in one runner generating test data and executing load test:
+```
+exampleconfigs/users/run.sh
+```
+
+What needs to be done to run it:
+* Create spanner instance
+* Create database named `users` 
+* Set `project` and `instance` in each of the config JSON files located under `exampleconfigs/users/users_config.json`
+    *`exampleconfigs/users/users_config.json`
+    *`exampleconfigs/users/groupmgt_config.json`
+    *`exampleconfigs/users/membership_config.json`
+    *`exampleconfigs/users/loadtestusers.json`
+* 
```

Comment:
this does not appear to format correctly. See https://github.com/googleapis/java-spanner-jdbc/blob/60a355aec261f0174c19688058b22d1a71e19492/samples/quickperf/readme.md#end-to-end-example

Reply:
fixed